api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -120,6 +120,14 @@
 
   String REWRITE_JOB_ORDER_DEFAULT = RewriteJobOrder.NONE.orderName();
 
+  /**
+   * Constant representing the target output partition specification ID.
+   *
+   * <p>This ID is used by the file rewriter during the rewrite operation to identify the specific
+   * output partition.
```

Comment:
partition -> "partition spec. This allows rewriting files into a partitioning which is not the current schema's partitioning. When not specified the current partitioning is used."

Or something like that?